pkg/daemon/ceph/client/filesystem_mirror.go

```diff
@@ -79,7 +79,12 @@ func DisableFilesystemSnapshotMirror(context *clusterd.Context, clusterInfo *Clu
 	// Run command
 	output, err := cmd.Run()
 	if err != nil {
-		return errors.Wrapf(err, "failed to disable ceph filesystem snapshot mirror for filesystem %q. %s", filesystem, output)
+		// We cannot wrap here since errors.Wrapf will return *errors.withStack which is not exported so we can't assert it
+		// The call reads *exec.ExitError so won't be work
+		if code, err := exec.ExtractExitCode(err); err == nil && code != int(syscall.ENOTSUP) {
```

Comment:
Instead of ignoring this error, is there a way we can avoid disable mirroring in the first place? Since mirroring wasn't ever enabled, the reconcile shouldn't be trying to disable it, right? I'd like to understand better what causes it during the upgrade when we don't see it in a new install.

Reply:
Mirroring should be disabled already. Even if we understand what went wrong during the upgrade this won't prevent users from editing the spec and making it non-null. Ignoring `ENOTSUP` seems like a valid solution to me and will avoid user mistakes.

Reply:
Ok, ignoring the error sounds fine. It just seems unexpected that it would not be nil in upgrades and be nil in a new install (and avoid hitting this error). The values look the same to me whether it was a new install or an upgrade, but I must be missing something.

Reply:
So the problem is that in 1.6.8 the type `FSMirroringSpec` was not a pointer so we were getting an empty type, not a nil pointer. That's why the check is inefficient during the upgrade. Also ideally `Enabled` could have been a pointer so we would know when mirroring is explicitly set or not.

I think sticking with the current approach is the best thing to do we avoid more complex handling in the code like `!= nil` AND now using reflection for deep equal against `cephv1.FSMirroringSpec{}`.